src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/CourtService.kt

```diff
@@ -62,4 +72,66 @@ class CourtService(
         )
       }.toSet()
   }
+
+  fun createVideoLinkBooking(specification: VideoLinkBookingSpecification): Long {
+    val mainId = savePrisonAppointment(specification, specification.main)
```

Comment:
How safe is this? Safe enough? 3 Remote REST calls create 3 appointment records in NOMIS.
If call 2 or 3 fails then record 1 and possibly 2 will be orphaned.  
Possible solutions: 
* Catch errors and attempt compensating deletes of orphaned records
* Create a new end-point in prison-api that accepts an array of appointments to create - transactional - all/none created
* Don't trust the client: Check that the locationIds exist before attempting to create the records. (unacceptable bookingId will fail on first create so isn't a problem for integrity)
Should this code check all the client parameters and enforce constraints? 
Examples
* Times: pre start < pre end <= main start < main end <= post start <= post end
* Locations are of the right kind ("VELB") and in the same agency as the bookingId
* bookingId is in an agency where VLB is enabled.
These checks are done in the front ends - BVL and whereabouts front end so not essential to repeat in this API - but... is this technical debt?

Reply:
I like the idea around doing some checks within the service. Also what I've done in an open PR for cell move stuff is do a fair bit of logging. The idea is I'd be able to see how far it got and what orphaned records would need to be deleted later on.